app/controllers/users_controller.rb

```diff
@@ -1,4 +1,12 @@
 class UsersController < ApplicationController
+  before_action :require_admin, only: [:index]
+
+  def index
+    @users = User.all
+
+    respond_with(@users)
+  end
```

Comment:
I would consider using admin namespace in routes, and have admin controllers for that. We can chat more about this next week. Maybe for another PR to refactor this out.

Reply:
OK, sounds good, I am interested to give this a go